ui/src/app/applications/components/application-details/application-details.tsx

```diff
@@ -180,6 +180,216 @@ export class ApplicationDetails extends React.Component<RouteComponentProps<{app
+                                    </div>
+                                    <div className='row white-box__details-row'>
+                                        <div className='columns small-3'>Chart Source:</div>
+                                        <div className='columns small-9'>{aSource.repoURL}</div>
+                                    </div>
+                                    <div className='row white-box__details-row'>
+                                        <div className='columns small-3'>Helm Chart:</div>
+                                        <div className='columns small-9'>
+                                            {aSource.chart}&nbsp;
+                                            {
+                                                <a
+                                                    title={sources[indx].chart}
+                                                    onClick={e => {
+                                                        e.stopPropagation();
+                                                        window.open(aSource.repoURL);
+                                                    }}>
+                                                    <i className='fa fa-external-link-alt' />
+                                                </a>
+                                            }
+                                        </div>
+                                    </div>
+                                </div>
+                            </div>
+                        );
+                    }}
+                </DataLoader>
+            );
+        };
+
+        const getContentForNonChart = (
+            aRevision: string,
+            aSourceIndex: number,
+            aVersionId: number,
+            indx: number,
+            aSource: models.ApplicationSource,
+            sourceHeader?: JSX.Element
+        ) => {
+            return (
+                <DataLoader
+                    key={indx}
+                    load={() => services.applications.revisionMetadata(application.metadata.name, application.metadata.namespace, aRevision, aSourceIndex, aVersionId)}>
+                    {metadata =>
+                        metadata ? (
+                            <div className='white-box' style={{marginTop: '1.5em'}}>
+                                {sourceHeader && sourceHeader}
+                                <div className='white-box__details'>
+                                    <div className='row white-box__details-row'>
+                                        <div className='columns small-3'>SHA:</div>
+                                        <div className='columns small-9'>
+                                            <Revision repoUrl={aSource.repoURL} revision={aRevision} />
+                                        </div>
+                                    </div>
+                                </div>
```

Comment:
@keithchong , this block looks quite redundant and used multiple times here. Do you think we could create a separate function and pass in the values that need to be displayed to it?

Reply:
Thanks @ishitasequeira , I've commoned up two sections.  BTW, the panels launched from the "..."  (f.k.a. More) are impacted, for both multi-source and single source apps